fecgov/openFEC
# Fix logic for full cycle candidate committee totals (Part 1)

## Summary (required)

Resolves https://github.com/fecgov/fec-cms/issues/2631 along with front-end PR (Part 2) : https://github.com/fecgov/fec-cms/pull/2635

- Fixes the full election cycle totals for 2020+ candidates in the gap between the end of the year and the first filings
**MV Changes:**
 - Round election year (line 39/66)
 - Use cand_election year instead of fec_election_yr (line 66)
 - Fix cycle for full election cycles (line 206)
 - Join election totals on election year, not cycle

## How to test the changes locally
- Connect to `dev` or `dev_replica` DB
- Compare `ofec_totals_candidate_committees_mv_lb_tmp` to current `ofec_totals_candidate_committees_mv`
- Change models `CandidateCommitteeTotalsPresidential` and `CandidateCommitteeTotalsHouseSenate` to point to temporary MV `ofec_totals_candidate_committees_mv_lb_tmp`
- Test migrations: drop/create `cfdm_test`, `invoke create_sample_db`
- Point local CMS (`feature/fix-2020-totals` branch) to local API
- Compare 2-year/all-year totals to production:
- 2020 senate: 
http://localhost:8000/data/candidate/S6IL00151 
http://localhost:8000/data/candidate/S4NJ00185
http://localhost:8000/data/candidate/S2TX00106
- 2022 senate:
http://localhost:8000/data/candidate/S4PA00121/
http://localhost:8000/data/candidate/S6CA00584/
- 2018/24 senate (check 2018 data):
http://localhost:8000/data/candidate/S8FL00273/?cycle=2018&election_full=true
http://localhost:8000/data/candidate/S8TX00285/
http://localhost:8000/data/candidate/S6MO00305/
- 2020 presidential:
http://localhost:8000/data/candidate/P80001571/
http://localhost:8000/data/candidate/P00006213/
http://localhost:8000/data/candidate/P00006486/
- 2018 House:  `H8CA05035` (2 year periods only - no 2020 data yet)
- Test local CMS: 2020 prez, 2020/2022/2024 senate, house,  historical data. Committee profile pages should be unchanged.

## Impacted areas of the application
List general components of the application that this PR will affect:

- Candidate profile pages "all-year" totals
- Raising/Spending tabs should appear on the left where there's financial data for 2018

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
feature/fix-2020-totals | https://github.com/fecgov/fec-cms/pull/2635
